samcli/hook_packages/terraform/hooks/prepare.py

```diff
@@ -718,15 +719,20 @@ def _get_python_command_name() -> str:
     str
         The name of the python command installed
     """
-    command_names_to_try = ["python", "python3"]
+    command_names_to_try = ["python3", "py3", "python", "py"]
     for command_name in command_names_to_try:
         try:
-            run([command_name, "--version"], check=True, capture_output=True)
+            run_result = run([command_name, "--version"], check=True, capture_output=True, text=True)
         except CalledProcessError:
             pass
         else:
+            # check for python 3, 3.7 or above
+            # regex: search for 'Python', whitespace, '3.', digits 7-9 or 2+ digits, any digit or '.' 0+ times
+            match = re.search(r"Python\s*3.([7-9]|\d{2,})[\d.]*", run_result.stdout)
```

Comment:
nit: this maybe ok in perf front since we have a limit on the command names to try, but in general its better to separate the match by creating a re.compile(expr) ahead of time and then re-using it.

you can benchmark things like this by using `timeit`

python -m timeit -s "<stmt>"

Reply:
It seems like due to caching (and like you mentioned, our small number of commands to try) this should be a negligible difference: https://stackoverflow.com/a/47269110

Reply:
made the change to pre-compile anyway ✅